lib/Sema/TypeCheckStmt.cpp

```diff
@@ -119,7 +119,12 @@ namespace {
 
       // Caller-side default arguments need their @autoclosures checked.
       if (auto *DAE = dyn_cast<DefaultArgumentExpr>(E))
-        if (DAE->isCallerSide() && DAE->getParamDecl()->isAutoClosure())
+        if (DAE->isCallerSide() &&
+            (DAE->getParamDecl()->isAutoClosure() ||
+             (DAE->getParamDecl()->getDefaultArgumentKind() ==
+                  DefaultArgumentKind::ExpressionMacro &&
+              ParentDC->getASTContext().LangOpts.hasFeature(
```

Comment:
I don't think we need this `hasFeature` check here. Checking at the definition site is good enough.